### PR TITLE
Refactor bash functions to use atlas-bash-util

### PR DESCRIPTION
**Pull request by pmb59**
In the repo there are functions that are already available here:
https://github.com/ebi-gene-expression-group/atlas-bash-util

The goal is to do some code refactoring to make use of atlas-bash-util. 

A submodule, rather than activating a new conda env in multiple rules, seems fine as mostly sourcing bash functions here and the dependencies of atlas-bash-utils are ensured (bash, coreutils, grep).

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -1090,6 +1087,7 @@ rule decorate_expression_baseline:
         set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
         exec &> "{log}"
         source {workflow.basedir}/bin/reprocessing_routines.sh
+        source {workflow.basedir}/atlas-bash-util/generic_routines.sh
```
Comment (pcm32) [inline Snakefile:1090]: if you are only adding a source here, how were these variables provisioned before? The ones that you are getting from bash-util/generic_routines?

File: Snakefile-sorting-hat [hunk 2]
```diff
@@ -165,7 +165,7 @@ rule get_experiment_metadata:
         export ATLAS_META_CONFIG={params.atlas_meta_config}
         export ATLAS_PROD={params.atlas_prod}
         export ATLAS_EXPS={params.atlas_exps}
-        source {workflow.basedir}/bin/metadata_retrieval_functions.sh
+        source {workflow.basedir}/atlas-bash-util/generic_routines.sh
```
Comment (pcm32) [inline Snakefile-sorting-hat:168]: My concern here is that metadata_retrieval_functions was very specific and generic routines is quite wide? Is this replacement working well?

File: bin/quantile_normalize.sh [hunk 3]
```diff
@@ -44,7 +44,7 @@ trap 'rm -f ${tmp_files}.*' INT TERM EXIT
 
 tmp_config_tsv=$tmp_files.config.tsv
 #"$scriptDir/get_assay_groups.pl" "$path_to_experiment_config" > "$tmp_config_tsv"
-"$scriptDir/get_assay_groups.pl" "$path_to_experiment_config" > "$tmp_config_tsv"
+"$scriptDir/atlas-bash-util/get_assay_groups.pl" "$path_to_experiment_config" > "$tmp_config_tsv"
```
Comment (pcm32) [inline bin/quantile_normalize.sh:47]: Are these identical? I remember having to add some changes to the version I brought to this repo. Or maybe it was another function.

**Review by pcm32 — APPROVED** [review]
Looks good in general, really biggest thing here is whether this all works when tried, I just added some cautionary comments, but if in your tests things are fine, then all good.

**Comment by pinin4fjords** [comment]
Could you clarify why it's necessary to use atlas-bash-util in this way rather than via the conda package? That makes all the contained functions easily accessible (like https://github.com/ebi-gene-expression-group/atlas-bash-util#generic-routines), without all the 'generic routines' imports you have here.

**Comment by pcm32** [comment]
On one side I don't mind the submodule so much since it makes development easier and this is a rather simple package. Otherwise we need to make a conda release each time, rebuild environments, etc.

**Comment by pcm32** [comment]
It is true though that this package should be fairly static.

File: Snakefile (hunk 1 above)
Comment (pmb59) [reply Snakefile:1090]: The variables are provisioned as before. Specifically (see below) for 
```
geneNameFile=$( get_geneNameFile_given_organism {params.organism}  )
```
`find_properties_file` now is sourced from atlas-bash-util/generic_routines.sh

File: Snakefile-sorting-hat (hunk 2 above)
Comment (pmb59) [reply Snakefile-sorting-hat:168]: I have checked that all the code being replaced is equal. It works

File: bin/quantile_normalize.sh (hunk 3 above)
Comment (pmb59) [reply bin/quantile_normalize.sh:47]: In this case, they are identical.

File: Snakefile (hunk 1 above)
Comment (pmb59) [reply Snakefile:1090]: I did, however, found another case where a variable needed to be exported [c2e555f](https://github.com/ebi-gene-expression-group/bulk-recalculations/pull/40/commits/c2e555f44c70b977d64ab14ee591da123e55be6f)

**Comment by pmb59** [comment]
> Could you clarify why it's necessary to use atlas-bash-util in this way rather than via the conda package? That makes all the contained functions easily accessible (like https://github.com/ebi-gene-expression-group/atlas-bash-util#generic-routines), without all the 'generic routines' imports you have here.

@pinin4fjords In this case, currently, a submodule is more of a benefit (as @pcm32 mentioned, it makes development easier) for this collection of bash functions, instead of using dependency management system. The submodule implies some imports but it is also true that arranging a self-contained conda env in several Snakamake rules also brings a certain computational cost associated to it.  Nevertheless, I will open an issue to consider moving to conda at later stage.

**Review by pcm32 — APPROVED** [review]
LGTM - I would add peach API as config, but since we are changing that very soon I'm not so concerned.

**Pull request merged**